src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ClientInterface.cs

```diff
@@ -0,0 +1,453 @@
+                        if (osHalf + runtimeHalf < remaining)
+                        {
+                            ++osHalf;
+                        }
+                        
+                        Debug.Assert(osHalf + runtimeHalf == remaining);
+                        
+                        // Will the OS Info fit as-is?
+                        if (osInfo.Length <= osHalf)
+                        {
+                            // Yes, so the Runtime Info must be too long.
+                            // Truncate it as little as possible.
+                            runtimeInfo =
+                                runtimeInfo.Substring(
+                                    0,
+                                    remaining - osInfo.Length);
+                        }
+                        // Will the Runtime Info fit as-is?
+                        else if (runtimeInfo.Length <= runtimeHalf)
+                        {
+                            // Yes, so the OS Info must be too long.  Truncate
+                            // it as little as possible.
+                            osInfo =
+                                osInfo.Substring(
+                                    0,
+                                    remaining - runtimeInfo.Length);
+                        }
+                        // Otherwise, we need to truncate them both.
+                        else
+                        {
+                            osInfo = osInfo.Substring(0, osHalf);
+                            runtimeInfo = runtimeInfo.Substring(0, runtimeHalf);
+                        }
+
+                        Debug.Assert(
+                            osInfo.Length + runtimeInfo.Length <= remaining);
+                    }
+
+                    // Append them now that they've been truncated if necessary.
+                    name.Append(osInfo);
+                    name.Append('|');
+                    name.Append(runtimeInfo);
+                }
+
+                // Remember the name we've built up.
+                result = name.ToString();
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                // StringBuilder failed in an unexpected way, so use our
+                // fallback value.
+                result =
+                    $"{driverName}|{Unknown}|{Unknown}|{Unknown}|{Unknown}";
+            }
+
+            // Truncate to our max length if necessary.
+            //
+            // This is a paranoia check to ensure we don't violate our API
+            // promise.
+            //
+            if (result.Length > maxLen)
+            {
+                // We know this won't throw ArgumentOutOfRangeException because
+                // we've already confirmed that Length is greater than maxLen.
+                result = result.Substring(0, maxLen);
+            }
+
+            return result;
+        }
+
+        // Clean the given value of any disallowed characters, replacing them
```

Comment:
For these comments on methods, it would be good to stick to /// kind of comments, so that we can leverage VS support of hovering over the method name and looking at the formatted comments.

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/